AxonDeepSeg/morphometrics/aggregate.py

```diff
@@ -0,0 +1,221 @@
+"""
+This script is used to aggregate morphometric data and compute basic statistics,
+such as mean g-ratio per subject. It also computes statistics per axon diameter 
+ranges (e.g. mean g-ratio for axons with diameter between 0.5 and 1 um, between 
+1 and 2, etc.). It assumes the following input directory structure:
+
+---
+input_dir
+└───subject1
+│   │   img_axon_morphometrics.xlsx
+|   |   img2_axon_morphometrics.xlsx
+│   │   ...
+└───subject2
+│   │   ...
+---
+
+The script will create an aggregate morphometrics file per subject and a global 
+one with all subjects, in a output directory called "morphometrics_agg".
+"""
+
+from pathlib import Path
+from loguru import logger
+import argparse
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+import os
+import math
+
+import AxonDeepSeg
+from AxonDeepSeg.params import morph_suffix, agg_dir
+
+AXON_MORPHOMETRICS = "axon_morphometrics"
+METRICS_FILE_NAME = "axon_metrics"
+METRICS_NAMES = {
+    ("Axon Diameter", "axon_diam (um)"),
+    ("G-Ratio", "gratio"),
+    ("Myelin Thickness", "myelin_thickness (um)"),
+}
+OUTPUT_FOLDER = "morphometrics_agg"
+
+
+def save_axon_count_plot(
+    subject_df, subject_name, subject_folder_name, labels, output_folder=OUTPUT_FOLDER
+):
+
+    morph_subject_path = os.path.join(output_folder, subject_folder_name, subject_name)
+    os.makedirs(morph_subject_path, exist_ok=True)
+    morph_figures_path = os.path.join(morph_subject_path, "figures_axon_count")
+    os.makedirs(morph_figures_path, exist_ok=True)
+
+    plt.figure(figsize=(8, 5))
+    sns.barplot(
+        data=subject_df, x="axon_bin", y="axon_diam (um)", estimator=len, order=labels
+    )
+    plt.title(f"Axon diameters for subject {subject_name}")
+    plt.xlabel("Axon Diameter (um)")
+    plt.ylabel("Count")
+
+    save_path = os.path.join(morph_figures_path, f"{subject_name}.png")
+    plt.savefig(save_path, dpi=300, bbox_inches="tight")
+
+    plt.show()
```

Comment:
plt.show() should be removed. It makes the script crash after a single iteration on my side.